src/courseware/course/sidebar/sidebars/course-outline/CourseOutlineTrigger.jsx

```diff
@@ -1,32 +1,56 @@
+import { useContext } from 'react';
+import classNames from 'classnames';
 import PropTypes from 'prop-types';
 import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import { IconButton } from '@openedx/paragon';
+import { MenuOpen as MenuOpenIcon } from '@openedx/paragon/icons';
 
-import SidebarTriggerBase from '../../common/TriggerBase';
 import { LAYOUT_LEFT } from '../../common/constants';
-import CourseOutlineIcon from './CourseOutlineIcon';
+import SidebarContext from '../../SidebarContext';
 import messages from './messages';
 
 export const ID = 'COURSE_OUTLINE';
 export const LAYOUT = LAYOUT_LEFT;
 
-const CourseOutlineTrigger = ({
-  intl,
-  onClick,
-}) => {
-  const handleClick = () => {
-    onClick();
-  };
+const CourseOutlineTrigger = ({ intl, isMobileView }) => {
+  const {
+    currentSidebar,
+    toggleSidebar,
+    shouldDisplayFullScreen,
+  } = useContext(SidebarContext);
+
+  const isDisplayForDesktopView = !isMobileView && !shouldDisplayFullScreen && currentSidebar !== ID;
+  const isDisplayForMobileView = isMobileView && shouldDisplayFullScreen;
+
+  if (!isDisplayForDesktopView && !isDisplayForMobileView) {
+    return null;
+  }
+
+  const handleToggleCollapse = () => (currentSidebar === ID ? toggleSidebar(null) : toggleSidebar(ID));
```

Comment:
```suggestion
  const handleToggleCollapse = () => currentSidebar === ID ? toggleSidebar(null) : toggleSidebar(ID);
```

Reply:
We will catch an eslint error `ESLint: Arrow function used ambiguously with a conditional expression.(no-confusing-arrow)`